packages/react/select/src/Select.tsx

```diff
@@ -233,14 +225,19 @@ const SelectTrigger = React.forwardRef<SelectTriggerElement, SelectTriggerProps>
         aria-controls={context.contentId}
         aria-expanded={context.open}
         aria-autocomplete="none"
-        aria-labelledby={labelledBy}
         dir={context.dir}
         data-state={context.open ? 'open' : 'closed'}
         disabled={disabled}
         data-disabled={disabled ? '' : undefined}
         data-placeholder={context.value === undefined ? '' : undefined}
         {...triggerProps}
         ref={composedRefs}
+        onClick={composeEventHandlers(triggerProps.onClick, (event) => {
+          // enable compatibility with native label or custom `Label` "click"
+          // this doesn't create any other side-effect because we are preventing default
+          // in `onPointerDown` so effectively this only runs for a label "click"
```

Comment:
My memory is already failing but was this a Safari specific compatibility? if so should we be more explicit about that here?

Reply:
Thanks, reads great 👍